src/components/CreateNewProjectWizard/Gatsby/SelectStarterDialog.js

```diff
@@ -0,0 +1,233 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import fetch from 'node-fetch'; // Note: This is using net.request from Node. Browser fetch throws CORS error.
+import styled from 'styled-components';
+import yaml from 'js-yaml';
+import Scrollbars from 'react-custom-scrollbars';
+import { Tooltip } from 'react-tippy';
+
+import Divider from '../../Divider';
+import Spacer from '../../Spacer';
+import ExternalLink from '../../ExternalLink';
+
+// import { SearchBox } from 'react-instantsearch/dom';
+// import {
+//   InstantSearch,
+//   InfiniteHits,
+//   Configure,
+// } from 'react-instantsearch/dom';
+
+import StrokeButton from '../../Button/StrokeButton';
+import Paragraph from '../../Paragraph';
+import Heading from '../../Heading';
+import Modal from '../../Modal';
+import ModalHeader from '../../ModalHeader';
+
+import * as actions from '../../../actions';
+
+// import { ALGOLIA_KEYS } from '../../../constants';
+import { COLORS } from '../../../constants';
+import type { Dispatch } from '../../../actions/types';
+
+type Props = {
+  updateFieldValue: (string, string) => void,
+  selectedStarter: string,
+  isVisible: boolean,
+  hideModal: Dispatch<typeof actions.hideStarterSelectionModal>,
+};
+
+type State = {
+  loading: boolean,
+  starters: Array<any>,
+  selectedStarterInModal: string,
+};
+
+/*
+Gatsby starter selection dialog.
+A starter object from starter.yml contains the following data:
+[
+   {
+      "url": "https://wonism.github.io/",
+      "repo": "https://github.com/wonism/gatsby-advanced-blog",
+      "description": "n/a",
+      "tags": [
+        "Portfolio",
+        "Redux"
+      ],
+      "features": [
+        "feature items"
+      ]
+  }, ... 
+]
+*/
+class SelectStarterDialog extends PureComponent<Props, State> {
+  state = {
+    loading: true,
+    starters: [],
+    selectedStarterInModal: '',
+  };
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    return {
+      selectedStarterInModal: nextProps.selectedStarter,
+    };
+  }
+
+  componentDidMount() {
+    fetch(
+      'https://raw.githubusercontent.com/gatsbyjs/gatsby/master/docs/starters.yml'

```

Comment:
to put this in Algolia, we will have to push to Algolia somewhere https://github.com/gatsbyjs/gatsby/blob/master/www/gatsby-node.js here I think, but I'm not too sure yet what the most efficient way will be.

Reply:
Thanks, for pointing to the Graphql query in the Gatsby starter. It would be great if we could use a similar query but I'm not sure how to query the same data.
I think for now we can just use the `starter.yml` and I think the selection is OK now. I'll update the screenshots above in a minute.

Also the search can be easily done by filtering through the yml data. So Algolia is not really needed here at the moment.